### PR TITLE
Bug 1871895: Use form data for application ImageStream name

### PR DESCRIPTION
**Pull request by rottencandy**
**Fixes**: 

https://issues.redhat.com/browse/ODC-4615
https://issues.redhat.com/browse/ODC-4649

**Analysis / Root cause**: 

The add flow tries to get image name from imagestream labels, which may not be always present.

**Solution Description**: 

Use application form data to get image stream name.

**Screen shots / Gifs for design review**: 

rec

**Unit test coverage report**: 

Passing.

**Browser conformance**: 

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts [hunk 1]
```diff
@@ -239,6 +239,7 @@ export const getUserLabels = (resource: K8sResourceKind) => {
     'app.kubernetes.io/part-of',
     'app.openshift.io/runtime-version',
     'app.openshift.io/runtime-namespace',
+    'app.openshift.io/runtime-icon',
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts:242]: Please revert this change to use `runtime` label. runtime label itself points the imagestream icon.
```suggestion
    'app.openshift.io/runtime',
```

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts (hunk 1 above)
Comment (rottencandy) [reply frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts]: Updated.

**Review by sahil143 — CHANGES_REQUESTED** [review]
Once in edit flow Image stream is not selected the first time dropdown is clicked.
deployimg

**Comment by rottencandy** [comment]
> /lgtm

@divyanshiGupta I'm still working on Sahil's comment.

**Comment by divyanshiGupta** [comment]
> > /lgtm
> 
> @divyanshiGupta I'm still working on Sahil's comment.

Oops missed that. Let me know once you are done.
/lgtm cancel

**Review by divyanshiGupta — APPROVED** [review]
/lgtm

File: frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx [hunk 2]
```diff
@@ -63,7 +63,14 @@ const ImageStreamNsDropdown: React.FC = () => {
   );
 
   React.useEffect(() => {
+    if (
+      initialValues.imageStream.image &&
+      values.imageStream.image !== initialValues.imageStream.image
+    ) {
+      initialValues.imageStream.image = values.imageStream.image;
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx:70]: resetting initial values seems odd @rohitkrai03 @divyanshiGupta thoughts?

Comment (divyanshiGupta) [reply frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx:70]: It's generally not recommended to change the initial values but here I think resetting the initial value is fine since we just need the initial value only when the form is opened.

Comment (rottencandy) [reply frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx:70]: This effect gets fired the first time `ImageStreamDropdown` is changed, it's meant to verify the current namespace permissions, but ends up resetting the selected ImageStream name back to the initial value.

**Comment by invincibleJai** [comment]
/approve

Verified the changes , works as expected

/cc @karthikjeeyar

**Comment by karthikjeeyar** [comment]
Verified locally and it works fine.
/lgtm
